components/service/experiments/src/main/java/mozilla/components/service/experiments/ExperimentsUpdater.kt

```diff
@@ -25,23 +29,55 @@ internal class ExperimentsUpdater(
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
     internal lateinit var source: KintoExperimentSource
 
-    @Volatile private var updateJob: Job? = null
```

Comment:
Per the previous comments, I will add async executions back to specific test(s) in order to test this.  Using WorkManager now, it may be that it has to be faked and awaited a little differently than before.